core/Pos/Binary/Merkle.hs

```diff
@@ -4,16 +4,16 @@ module Pos.Binary.Merkle () where
 
 import           Universum
 
-import           Pos.Binary.Class   (Bi (..), Raw, label, putField)
+import           Pos.Binary.Class   (Bi (..), Raw)
 import           Pos.Crypto.Hashing (Hash)
 import           Pos.Merkle         (MerkleRoot (..), MerkleTree (..), mkMerkleTree)
 
 -- This instance is both faster and more space-efficient (as confirmed by a
 -- benchmark). Hashing turns out to be faster than decoding extra data.
 instance (Bi a, Bi (Hash Raw)) => Bi (MerkleTree a) where
-    sizeNPut = putField toList
-    get = label "MerkleTree" $ mkMerkleTree <$> get
+  encode = encode . toList
+  decode = mkMerkleTree <$> decode
 
 instance (Bi a, Bi (Hash Raw)) => Bi (MerkleRoot a) where
-    sizeNPut = putField getMerkleRoot
-    get = label "MerkleRoot" $ MerkleRoot <$> get
+  encode = encode . getMerkleRoot
+  decode = MerkleRoot <$> decode
```

Comment:
On the other hand, I'm not sure that it won't mess with other people's Emacs settings. Hmmmmm. At any rate, for now you can create the file locally:

``` elisp
;;; Directory Local Variables
;;; For more information see (info "(emacs) Directory Variables")

((haskell-mode
  (haskell-indentation-starter-offset . 4)
  (haskell-indentation-left-offset . 4)
  (haskell-indentation-layout-offset . 4)
  (haskell-indent-spaces . 4)
  (haskell-stylish-on-save . t)))
```